### PR TITLE
fix(atomic): fix facet search scrolling & focus issues

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-629

(When I mean search & search box here, it's for the facet, not the atomic-search-box search)

**To recap:**
- Removed "show more button" in the facet search results
- We show more depending on the scroll position, the more we go down in the list, the more we fetch (like JSUI)
- The selected element is now always shown in scroll view when using the keyboard
- The selected element is reset when changing the query
- The query is now cleared when leaving the search box (like JSUI)
- We don't trigger a new search when selecting an element
- We hide the result when it's querying (loading) I'M NOT A SUPER FAN OF THIS ONE, [see my other comment](https://github.com/coveo/ui-kit/pull/730/files#r617012250)

### CODE REVIEW

File: packages/atomic/src/components/facets/facet-search/facet-search.pcss [hunk 1]
```diff
@@ -3,7 +3,7 @@
 }
 
 .search-results {
-  @apply z-10 absolute w-full bg-background border-divider-t-0 empty:border-none rounded-b overflow-y-scroll left-0;
+  @apply z-10 absolute w-full bg-background border-divider-t-0 empty:border-none rounded-b overflow-y-auto left-0;
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets/facet-search/facet-search.pcss:6]: On mac, the scrollbar is not as visible by default OS settings, it only appears when you actually scroll.

File: packages/atomic/src/components/facets/facet-search/facet-search.tsx [hunk 2]
```diff
@@ -184,15 +180,16 @@ export class FacetSearch {
   }
 
   private get searchResults() {
-    const showResults = this.component.showFacetSearchResults;
+    const showResults =
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets/facet-search/facet-search.tsx:183]: I think, like @kaitlynkatz previously mentioned, that we need some kind of clear search in Headless, that would empty the query & the results without triggering search.
The problem I had here is "showFacetSearchResults" would show on focus but still display the older results (if I used the search before) meanwhile it was fetching the newer results.

I could work around this, by using more local variables or hiding on isLoading, but I rather add a clean method in Headless to allow this.

Comment (samisayegh) [reply packages/atomic/src/components/facets/facet-search/facet-search.tsx:196]: Sounds good, we can add a `clear` method.

I just want to clarify the difference between two similar but slightly different UX interactions though:

1. If on blur text is not cleared

This is typically the case with the atomic search box. I would encourage people to hide the query suggest dropdown using CSS. When a user refocuses on the search box, the query suggestions will still be in state. There is no need to refetch them, saving a query and removing the lag to show the suggestions.

2. If on blur text is cleared

It makes sense to also clear the results. Providing a function that clears both text and results would definitely be helpful.

**Review by samisayegh — APPROVED** [review]

**Comment by ThibodeauJF** [comment]
@samisayegh for the facet search, if we follow the JSUI ux work for it, we clear on blur (option 2)
But wether we do 1 or 2, the same issue still appears after selecting a result:
- input query e.g. `BBC `
- select result
- visually hide values and clear text (but don't retrigger a new search, which is useless)
- when focusing back on the search box, we visually show back the results, but until the new results are loaded we see still the `BBC` results, a headless clear would have emptied the response on the state instead

Wether we do 1 or 2, we can't be sure that the results would still be the same, other parameters could have changed, so its safer to clear the result and redo a new search

File: packages/atomic/src/components/facets/facet-search/facet-search.tsx [hunk 3]
```diff
@@ -90,19 +82,41 @@ export class FacetSearch {
...
-      this.facetSearchController.search();
+    this.triggerSearch();
+  }
+
+  private onBlur() {
+    this.component.showFacetSearchResults = false;
+    this.text = '';
+  }
+
+  private onValuesScroll() {
```
Comment (olamothe) [inline packages/atomic/src/components/facets/facet-search/facet-search.tsx:112]: nitpick: just `onScroll`

**Review by olamothe — APPROVED** [review]
What is the plan for testing around this ?

**Comment by ThibodeauJF** [comment]
@olamothe tests will be on a separate PR, pointing to this one

File: packages/atomic/src/components/facets/facet-search/facet-search.tsx (hunk 2 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/facets/facet-search/facet-search.tsx:196]: (moved the comment here)

For the facet search, if we follow the JSUI ux work for it, we clear on blur (option 2)
But whether we do 1 or 2, the same issue still appears after selecting a result:
- input query e.g. `BBC `
- select result
- visually hide values and clear text (but don't retrigger a new search, which is useless)
- when focusing back on the search box, we visually show back the results, but until the new results are loaded we see still the `BBC` results, a headless clear would have previously emptied the values in the state.

Whether we do 1 or 2, we can't be sure that the results would still be the same, other parameters could have changed, so its safer to clear the result and redo a new search

**Comment by samisayegh** [comment]
@ThibodeauJF, I agree that the facet search UX falls under the second case. A `clear` method exposed by headless would be helpful,

**Comment by btaillon** [comment]
have you tested with mobile accessibility tools?

**Comment by ThibodeauJF** [comment]
Checked with talkback and while I'm not 100% sure how its used for combobox, when i swipe it appears to scroll down correctly, and more results load when it scrolls down

**Pull request merged**